src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java

```diff
@@ -40,32 +51,121 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
+                                                          String userAuthorisation) {
 
         CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         SscsCaseData sscsCaseData = caseDetails.getCaseData();
 
         PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
-
         PostponementRequest postponementRequest = sscsCaseData.getPostponementRequest();
-        String details = postponementRequest.getPostponementRequestDetails();
-        if (postponementRequest.getActionPostponementRequestSelected().equals(SEND_TO_JUDGE.getValue())) {
-            if (sscsCaseData.getAppealNotePad() == null) {
-                sscsCaseData.setAppealNotePad(NotePad.builder().notesCollection(new ArrayList<>()).build());
-            }
-            sscsCaseData.getAppealNotePad().getNotesCollection()
-                    .add(createPostponementRequestNote(userAuthorisation, details));
-            sscsCaseData.setInterlocReviewState(InterlocReviewState.REVIEW_BY_JUDGE.getId());
-            sscsCaseData.setInterlocReferralReason(InterlocReferralReason.REVIEW_POSTPONEMENT_REQUEST.getId());
-            sscsCaseData.setPostponementRequest(PostponementRequest.builder().unprocessedPostponementRequest(YesNo.YES).build());
+
+        if (isSendToJudge(postponementRequest)) {
+            sendToJudge(userAuthorisation, sscsCaseData);
+        } else if (isGrantPostponement(postponementRequest)) {
+            grantPostponement(sscsCaseData, postponementRequest);
+        } else if (isRefusePostponement(postponementRequest)) {
+            clearInterlocAndSetFlags(sscsCaseData);
         }
 
+        clearTransientFields(sscsCaseData);
         return response;
     }
 
+    private void grantPostponement(SscsCaseData sscsCaseData, PostponementRequest postponementRequest) {
+
+        if (isReadyToList(postponementRequest)) {
+            sscsCaseData.setState(State.READY_TO_LIST);
+        } else if (isNotListable(postponementRequest)) {
+            sscsCaseData.setState(State.NOT_LISTABLE);
+        }
+
+        updatePartyUnavailability(sscsCaseData);
+        clearInterlocAndSetFlags(sscsCaseData);
+    }
+
+    private void clearInterlocAndSetFlags(SscsCaseData sscsCaseData) {
+        sscsCaseData.setInterlocReferralReason(null);
+        sscsCaseData.setInterlocReviewState(null);
+        sscsCaseData.setDwpState(DwpState.DIRECTION_ACTION_REQUIRED.getId());
+        addDirectionNotice(sscsCaseData);
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().unprocessedPostponementRequest(YesNo.NO)
+                .build());
+    }
+
+    private void updatePartyUnavailability(SscsCaseData sscsCaseData) {
+        Optional<Hearing> futureHearing = sscsCaseData.getHearings().stream()
+                .filter(hearing -> LocalDate.parse(hearing.getValue().getHearingDate()).isAfter(LocalDate.now()))
+                .findAny();
```

Comment:
How does this work when there are multiple hearings in future?

Reply:
there should be only one hearing in the future according to Femi